### ISSUE TITLE
Missing resources fix mods causing CTD on jailbreaks.

### ISSUE THREAD

**Issue by jmartinez989** [long-war-2/lwotc#81]
[jailbreak.zip](https://github.com/long-war-2/lwotc/files/3067122/jailbreak.zip)
So I am in a point in the campaign where I am about to start a jailbreak mission but it always crashes to desktop as it starts loading. I have attached the save file in strategy where the mission is about to finish downloading.

I do have a list of mods running with LWotC but most are cosmetic and have worked up to this point. 2 are missing packages fixes that were for maps so maybe that can have something to do with it. I can provide the list of mods I have if needed.

**Comment by jmartinez989** [comment]
Ok so this seems to happen on all jailbreaks. I was able to queue up another and that one crashes too. Could it be possible it has something to do with the resistance soldiers? Not sure what could cause it.

**Comment by pledbrook** [comment]
This worries me. A bug with the exact same symptoms was [fixed](https://github.com/long-war-2/lwotc/commit/c1ba8a9c826386ed6ad4fc0a3cf381e55fb0c19a) well before the alpha-1 release. I'm hoping that this is just an environmental issue. Note that it does require changes to the community highlander, so make sure you don't have a copy of that enabled.

@ansabalirai @Tylon01 Have either of you had problems with the jailbreaks, AKA Rescue Resistance Contacts?

**Comment by ansabalirai** [comment]
None for me, but I am not running any mods adding new PCPs etc. Only did maybe 2 jailbreaks though. @jmartinez989 Can you try disabling as many mods as you can and trying if the issue still occurs.

**Comment by Tylon01** [comment]
I also have successfully completed a few jailbreak missions.  The only additional mod I am using is Evac All.

**Comment by MagSigeasol** [comment]
I undertook a jailbreak with no issues. No mods, no anarchy’s children

**Comment by pledbrook** [comment]
@jmartinez989 I managed to load that file successfully despite not having all of those mods. I suspect one or more of them are causing issues. Make sure none of them depend on/require the community highlander (although I don't think that should be a problem).

Note that Jailbreak is special in its objectives, which is why it required a change to the community highlander.

**Comment by jmartinez989** [comment]
Ok so it looks like I figured out which mod is the culprit here. It would seem that the mod that is causing the issue is WOTC Additional Icons. Might have something to do with how it interacts with the resistance fighters. Guess this one ca be closed. A shame though as I would like to use that mod.

I am going to look into using the tacitcal information overhaul. I states that you need the highlander for it but if it is integrated into LWotC then it should work so I will test it.

**Comment by pledbrook** [comment]
Interesting. I did briefly try out Additional Icons. If there's enough demand we could look into making it work with LW2 in the future.

**Comment by SirMrMister** [comment]
I think New Target Icons is the latest iteration (aka the most icons) for this kind of thing, if you wanna do it.

**Comment by jmartinez989** [comment]
I tried tactical information overhaul and it too crashes to desktop when loading a jailbreak. I do think it might be with how those mods interact with resistance fighters. It does work with other missions though.

In using it I actually like it better than additional icons as it looks cleaner and also makes it so the lifebar is always the same size instead of showing 1 square for each point of health making the healthbars of sectopods looks ridiculous.

I would def prefer Tactical Info Overhaul over Additional Icons. Also even if it only just crashes on jailbreaks I don't have to much of a problem with it. I would just save before a jailbreak, disable the mod and then start the jailbreak.

**Comment by jmartinez989** [comment]
Ok... so this can be closed. It is not TIO or Additional Icons that is the culprit. It is the mods Missing Packages Fix - Parcels and Maps Edition and Missing Packages Fix + Resource. Must have something to do with how the maps for jailbreaks load.

I was somehow able to load the jailbreak the first time by only disabling Additional Icons but then when I tried again it crashed.

I found this by turning mods on and off until I found it was those 2 causing the crash consistently. I am currently loaded into the jailbreak as I type this with TIO active.

**Comment by pledbrook** [comment]
I hate to say this, but are you sure? I have those mods enabled and was able to load your save game.

**Comment by jmartinez989** [comment]
well I did try loading again with both of them on and it crashed. Definitely those 2 that seem to cause it to crash the most consistently. I got to a point where even with TIO or additional icons turned off it was still crashing. That is when I had to go one by one.

It's possible there are certain map combinations in jailbreaks that wont crash with those 2 mods but most of the times I tried to load one it crashed.

**Comment by SirMrMister** [comment]
One really shouldn't use other mods (beside the four listed in the instructions) when playtesting, no matter how convenient they are. It'll only result in headeaches.

**Comment by jmartinez989** [comment]
I definitely wont be using any other new mods from here and if need be I'll just turn all but LWotC off. It's up to pledbrook if he wants to close this one but I wouldn't worry too much about it since at this point we know it's another mod causing this issue.

I have loaded more jailbreaks at this point without issue when turning off the missing packages mods so there's that.

**Comment by pledbrook** [comment]
Hmmm...I'll keep this open because I was considering making those missing resource mods requirements, because LW2 is missing quite a few assets. Admittedly, it's missing a bunch that aren't provided by those mods, so it may not be a big deal to leave them out.

**Comment by SirMrMister** [comment]
Regardless, it is a useful find that those two mods conflict with LWOTC. I suspect that since LWOTC does add new maps and missions, that it already has a partial workaround for at least what the Missing Parcels and Maps does. The crash occurring on a Jailbreak map must be no coincidence, since it might be a map based on assets that would now exist in duplicate or some shit if you load the mod.

**Comment by jmartinez989** [comment]
I changed the name of the  issue again to reflect what is actually going on since it is not actually related to TIO.

**Comment by pledbrook** [comment]
Huh. I may just have been bitten by this. Perhaps it depends on the map for the mission.

**Comment by rrnwang** [comment]
I would like to add a note for what I noticed seems to be a CTD during the final animation to kill an enemy:

`
[6403.65] ScriptLog: Active Objective: 
[6403.65] ScriptLog: Marked mission objective #0 as completed.
[6403.65] ScriptLog: (X2TacticalGameRuleset_2) X2TacticalGameRuleset::TurnPhase_UnitActions:EndBattle VictoriousPlayer:'XGPlayer_2' UILoseType:'eUICombatLose_FailableGeneric' GenerateReplaySave:'False'
[6403.65] ScriptLog: DLC2 OnTacticalGameEnd
[6403.65] ScriptLog: XComGameState_AchievementTracker.OnTacticalGameEnd
[6403.65] Critical: appError called: Assertion failed: appErrorf [File:G:\BuildAgent\work\9a884cb2af69f6ff\main\XCOM2\Development\Src\Core\Src\UnObj.cpp] [Line: 2456]
Failed to find function OnTacticalGameEnd in XComGameState_Effect_TemporaryItem Transient.XComEngine_0:XComGameStateHistory_0.XComGameState_464967.XComGameState_Effect_TemporaryItem_4
`

I noticed something is that it looks for some missing package. Additionally, it look for it in drive G. I do not have a G drive..
